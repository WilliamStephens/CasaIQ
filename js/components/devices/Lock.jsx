import React from 'react';
import PropTypes from 'prop-types';
import H3 from './H3';

const getCurrentLockState = (deviceID, deviceInfo) => {
  const lock = deviceInfo[deviceID];
  return lock[lock.length - 1].state === 'locked' ? 'unlock' : 'lock';
};

const buttonStyle = {
  margin: '0.5em',
};

const Lock = ({ deviceNameParts, deviceInfo, onLockClick }) => {
  const deviceID = deviceNameParts.join('-');
  const lockState = getCurrentLockState(deviceID, deviceInfo);
  return (
    <div className="box has-text-centered">
      <H3 deviceNameParts={deviceNameParts} />
      <button style={buttonStyle} className={`button ${lockState === 'lock' ? 'is-danger' : 'is-success'}`} type="button" onClick={() => onLockClick(deviceID)}>
        {lockState}
      </button>
    </div>
  );
};

Lock.propTypes = {
  deviceNameParts: PropTypes.arrayOf(PropTypes.string).isRequired,
  deviceInfo: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.shape({
    state: PropTypes.string.isRequired,
    timestamp: PropTypes.number.isRequired,
  }))).isRequired,
  onLockClick: PropTypes.func.isRequired,
};

export default Lock;
