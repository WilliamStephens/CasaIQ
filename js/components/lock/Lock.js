import React from "react"
import { captilizeFirstLetter } from "../../utilities/string";

const getCurrentLockState = (deviceName, locks) => {
    const lock = locks[deviceName];    
    return lock[lock.length - 1].state === "locked" ? "unlock" : "lock";
}

const getDeviceNameParts = deviceName => deviceName.split("-") 

const Lock = props => {
    const deviceNameParts = getDeviceNameParts(props.match.params.deviceName);
    console.log(props.locks)
    return (    
        <div>
            <h3>{`${captilizeFirstLetter(deviceNameParts[0])} ${deviceNameParts[1]} ${captilizeFirstLetter(deviceNameParts[2])}`}
            </h3>
            <div>
                <button type="button" onClick={() => props.onLockClick(props.match.params.deviceName)}>
                {getCurrentLockState(props.match.params.deviceName, props.locks)}
                </button>
            </div>      
        </div>
    )
}

export default Lock