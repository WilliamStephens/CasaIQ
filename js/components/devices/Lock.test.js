import renderer from 'react-test-renderer';
import React from 'react';
import Lock from './Lock';

test('Lock render', () => {
  const component = renderer.create(<Lock
    deviceNameParts={['apt', '145', 'lock']}
    deviceInfo={{ 'apt-145-lock': [{ state: 'locked', timestamp: 123 }] }}
    onLockClick={() => console.log('clicked')} // eslint-disable-line no-console
  />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
