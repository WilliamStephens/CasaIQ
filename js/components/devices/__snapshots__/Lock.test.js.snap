// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lock render 1`] = `
<div
  className="box has-text-centered"
>
  <div>
    <h3
      className="title"
    >
      Apt 145 Lock
    </h3>
  </div>
  <button
    className="button is-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "0.5em",
      }
    }
    type="button"
  >
    unlock
  </button>
</div>
`;
